tests/README.md

```diff
@@ -1,5 +1,29 @@
 # Devfile Registry Testing
 
+## Dependency check
+
+### Prerequisites
+
+- Ensure `yq 4.3x` is installed
+
+### Running the build script
+
+- This script performs three actions
+  - Clones samples from provided `extraDevfileEntries.yaml` under `samples/.cache`
+  - Creates a `parents.yaml` which contains the dependency tree for parent stacks
+  - Outputs the child sample paths of parent stacks, `TEST_DELTA=true` will result in only outputting child samples which have changed parent stacks
+- The build script takes one optional argument and works off of the current working directory
+  - `bash tests/build_parents_file.sh`, default samples file is `extraDevfileEntries.yaml`
+  - `bash tests/build_parents_file.sh <path_to_extraDevfileEntries>`
+
+### Use with testing
+
+- One can test the child samples using the [validate_devfile_schemas]() test suite by performing the following:
+```sh
+export STACKS=$(bash tests/build_parents_file.sh)
+STACKS_DIR=samples/.cache bash tests/validate_devfile_schemas.sh
```

Comment:
I've tried to test the script locally following the instructions:

- Upon root dir of project I had this error:
```
$ STACKS_DIR=samples/.cache bash tests/validate_devfile_schemas.sh

+ stacksDir=samples/.cache
+ stackDirs='nodejs-basic code-with-quarkus go-basic java-springboot-basic python-basic'
+ ginkgo run --procs 2 tests/validate_devfile_schemas -- -stacksPath samples/.cache -stackDirs 'nodejs-basic code-with-quarkus go-basic java-springboot-basic python-basic'
Failed to compile validate_devfile_schemas:

go: cannot find main module, but found .git/config in <mypath>/michael_valdron/devfile-registry
        to create a module there, run:
        go mod init

Ginkgo ran 1 suite in 12.799434ms
Test Suite Failed
```

Reply:
Made changes https://github.com/devfile/registry/pull/195/commits/7c0d08c3780917e6905f526be5cb5c6b619d62bb so `STACKS_DIR` can be either a relative or absolute path when executing `validate_devfile_schemas.sh`.

Reply:
> @thepetk What is your output for `go version`?

@michael-valdron output is
`go version go1.19.1 linux/amd64`

Reply:
> > @thepetk What is your output for `go version`?
> 
> @michael-valdron output is `go version go1.19.1 linux/amd64`

@thepetk try running it again with a go 1.18.x runtime, go 1.19.1 could be having issues running it.

Reply:
Finally got it run successfully :)

One thing I had to do other than those steps, was to add a `go.work` file in my repo's root dir.

It seems that is working fine :)

One test I tried to make is to update the go-basic sample inside the extraDevfileEntries. I've tried to include multiple sample versions:
```golang
  - name: go-basic
    displayName: Basic Go
    description: Go 1.16 application
    icon: https://go.dev/blog/go-brand/Go-Logo/SVG/Go-Logo_Blue.svg
    tags:
      - Go
    projectType: Go
    language: Go
    provider: Red Hat
    versions:
      - version: 1.0.2
        schemaVersion: 2.1.0
        git:
          remotes:
            origin: https://github.com/devfile-samples/devfile-sample-go-basic.git
      - version: 2.1.0
        schemaVersion: 2.2.0
        default: true
        git:
          checkoutFrom:
            revision: update_devfile
          remotes:
            origin: https://github.com/thepetk/devfile-sample-go-basic.git
```

In the parent file I had:
```
  - name: go
    children:
      - name: go-basic
        version: 1.0.2
  - name: go
    version: 2.1.0
    children:
      - name: go-basic
        version: 2.1.0
```
(Which is ok)

During testing I was expecting to see the test fail for one of the golang children (go-basic version 2.1.0). Although I had an issue with the `quarkus`. The trace of the test was:
```
• [FAILED] [1.984 seconds]
validate stacks follow the schema [It] stack: code-with-quarkus
/home/tpetkos/github/michael_valdron/devfile-registry/tests/validate_devfile_schemas/validate_devfile_schemas_test.go:50

  Timeline >>
  {/home/tpetkos/github/michael_valdron/devfile-registry/samples/.cache/go-basic/2.1.0/devfile.yaml  [] <nil> <nil> []  <nil> <nil> map[] <nil>}
  [FAILED] in [It] - /home/tpetkos/github/michael_valdron/devfile-registry/tests/validate_devfile_schemas/validate_devfile_schemas_test.go:61 @ 08/01/23 17:18:55.812
  << Timeline

  [FAILED] Expected
      <*errors.withStack | 0xc0004dde60>: {
          error: <*errors.withMessage | 0xc000693e40>{
              cause: <*multierror.Error | 0xc000693e20>{
                  Errors: [
                      <*errors.errorString | 0xc000450cd0>{
                          s: "Some Commands are already defined in parent: build-image, deploy, deployk8s. If you want to override them, you should do it in the parent scope.",
                      },
                  ],
                  ErrorFormat: nil,
              },
              msg: "failed to populateAndParseDevfile",
          },
          stack: [0x14db668, 0x14e2e38, 0x156ad58, 0x1530e1b, 0x1543958, 0x46d841],
      }
  to be nil
  In [It] at: /home/tpetkos/github/michael_valdron/devfile-registry/tests/validate_devfile_schemas/validate_devfile_schemas_test.go:61 @ 08/01/23 17:18:55.812
```

Reply:
>  During testing I was expecting to see the test fail for one of the golang children (go-basic version 2.1.0). 
> Although I had an issue with the quarkus. The trace of the test was:
> 
>```
> • [FAILED] [1.984 seconds]
> validate stacks follow the schema [It] stack: code-with-quarkus
> /home/tpetkos/github/michael_valdron/devfile-registry/tests/validate_devfile_schemas/validate_devfile_schemas_test.go:50
> 
>   Timeline >>
>   {/home/tpetkos/github/michael_valdron/devfile-registry/samples/.cache/go-basic/2.1.0/devfile.yaml  [] <nil> <nil> []  <nil> <nil> map[] <nil>}
>   [FAILED] in [It] - /home/tpetkos/github/michael_valdron/devfile-registry/tests/validate_devfile_schemas/validate_devfile_schemas_test.go:61 @ 08/01/23 17:18:55.812
>   << Timeline
> 
>   [FAILED] Expected
>       <*errors.withStack | 0xc0004dde60>: {
>           error: <*errors.withMessage | 0xc000693e40>{
>               cause: <*multierror.Error | 0xc000693e20>{
>                   Errors: [
>                       <*errors.errorString | 0xc000450cd0>{
>                           s: "Some Commands are already defined in parent: build-image, deploy, deployk8s. If you want to override them, you should do it in the parent scope.",
>                       },
>                   ],
>                   ErrorFormat: nil,
>               },
>               msg: "failed to populateAndParseDevfile",
>           },
>           stack: [0x14db668, 0x14e2e38, 0x156ad58, 0x1530e1b, 0x1543958, 0x46d841],
>       }
>   to be nil
>   In [It] at: /home/tpetkos/github/michael_valdron/devfile-registry/tests/validate_devfile_schemas/validate_devfile_schemas_test.go:61 @ 08/01/23 17:18:55.812
>```
>

This is also an interesting result, it is reporting the stack as being `code-with-quarkus` but the error in the test seems to be the expected one with `go-basic 2.1.0`, I'll need to check if there is a problem with the inputs or if there is a bug within this test suite which is causing the wrong stack name to be reported.